specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,649 @@
+                    newMenuItem->add_CustomItemSelected(Callback<ICoreWebView2CustomItemSelectedEventHandler>(
+                        [this, info](
+                            ICoreWebView2* sender,
+                            IUnknown* args)
+                            {
+                                wil::unique_cotaskmem_string linkUrl;
+                                info->get_LinkUrl(&linkUrl);
+                                m_appWindow->RunAsync([this, linkUrl]()
+                                {
+                                MessageBox(
+                                    m_appWindow->GetMainWindow(), linkUrl,
+                                    L"Display Link", MB_OK);
+                                });
+                                return S_OK;
+                            })
+                            .Get(),
+                    &m_customItemSelectedToken);
+                    CHECK_FAILURE(items->AddValueAtIndex(itemsCount, newMenuItem.get()));
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_contextMenuRequestedToken);
+```
+
+## Win32 C++ Use Data to Display Custom Context Menu
+
+The developer can use the data provided in the Event arguments to display a custom context menu with entries of their choice. For this case, the developer specifies Handled to be true and requests a deferral. Deferral of this event should be completed when the user selects a context menu item (either the app developer will handle the case, or can return the selected option to the WebView) or when they click on the screen (effectively closing the menu).
+
+ ```cpp
+    webview2->add_ContextMenuRequested(
+        Callback<ICoreWebView2ContextMenuRequestedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2ContextMenuRequestedEventArgs* args)
+            {
+                auto showMenu = [this, args]
+                {
+                    wil::com_ptr<ICoreWebView2ContextMenuItemCollection> items;
+                    CHECK_FAILURE(args->get_MenuItems(&items));
+                    args->put_Handled(true);
+
+                    UINT32 itemsCount;
+                    CHECK_FAILURE(items->get_Count(&itemsCount));
+
+                    HMENU hPopupMenu = CreatePopupMenu();
+                    wil::com_ptr<ICoreWebView2ContextMenuItem> current;
+
+                    for (UINT32 i = 0; i < itemsCount; i++) 
+                    {
+                        CHECK_FAILURE(items->GetValueAtIndex(i, &current));
+                        wil::unique_cotaskmem_string name;
+                        CHECK_FAILURE(current->get_Name(&name));
+                        UINT32 command_id;
+                        CHECK_FAILURE(current->get_CommandId(&command_id));
```

Comment:
The command ID is the actual edge-defined context menu item id that is used to invoke the command on the server side. I originally did have it where they could return the index, the problem that I faced was with the submenu items and how to use the index there. My solution was to use the command ID because it is unique and already a property of the server side context menu item, but would love to hear your thoughts.